src/App.tsx

```diff
@@ -1,26 +1,89 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID, getCount, getNewTodoId, getTodos } from './api/todos';
+import { Todo, Status } from './types/Todo';
+import { TodoList } from './components/todoList';
+import { Footer } from './components/footer';
+import { ErrorNotifications } from './components/errorNotifications';
+import { Header } from './components/header';
 
 export const App: React.FC = () => {
+  const [preparedTodos, setPreparedTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [selectedFilter, setSelectedFilter] = useState<string>(Status.all);
+  const [title, setTitle] = useState<string>('');
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState<number | null>(null);
+
+  if (errorMessage) {
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  }
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => {
+        switch (selectedFilter) {
+          case Status.active:
+            setPreparedTodos(todosFromServer.filter(todo => !todo.completed));
+            break;
+          case Status.complited:
+            setPreparedTodos(todosFromServer.filter(todo => todo.completed));
+            break;
+          default:
+            setPreparedTodos(todosFromServer);
+        }
+      })
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, [isSubmitting, selectedFilter, isLoading]);
```

Comment:
Better to fetch todos one time set it into state, and filter this state. It will improve performance and make code more secure